packages/cli-kit/src/public/node/ruby.test.ts

```diff
@@ -47,138 +54,176 @@ describe('execCLI', () => {
+    await file.inTemporaryDirectory(async (cli2Directory) => {
+      // Given
+      const execSpy = mockBundledCLI2(cli2Directory, !isWindows)
+      const gemfilePath = joinPath(cli2Directory, 'ruby-cli', RubyCLIVersion, 'Gemfile')
+
+      // When
+      await execCLI2(['args'], {
+        token: 'token_0000_1111_2222_3333',
+        directory: './directory',
+      })
+
+      // Then
+      validateBudleExec(execSpy, gemfilePath)
+      await validateGemFileContent(gemfilePath, isBundled, !isWindows)
     })
-
-    // Teardown
-    process.env = originalEnv
   })
 
-  it('run the CLI2 stored at a manual path', async () => {
-    // Setup
-    const originalEnv = process.env
-
-    // Given
-    const execSpy = vi.spyOn(system, 'exec')
-
-    process.env = {...originalEnv, SHOPIFY_CLI_2_0_DIRECTORY: '/manual', SHOPIFY_RUN_AS_USER: '1'}
-
-    vi.mocked(file.fileExists).mockResolvedValue(true)
-    vi.mocked(captureOutput).mockResolvedValueOnce('2.7.5')
-    vi.mocked(captureOutput).mockResolvedValueOnce('2.4.0')
-
-    // When
-    await execCLI2(['args'], {
-      token: 'token_0000_1111_2222_3333',
-      directory: './directory',
+  it('when run bundled CLI2 in windows then gemfile content should be correct and bundle runs with correct params', async () => {
+    await file.inTemporaryDirectory(async (cli2Directory) => {
+      // Given
+      const execSpy = mockBundledCLI2(cli2Directory, isWindows)
+      const gemfilePath = joinPath(cli2Directory, 'ruby-cli', RubyCLIVersion, 'Gemfile')
+
+      // When
+      await execCLI2(['args'], {
+        token: 'token_0000_1111_2222_3333',
+        directory: './directory',
+      })
+
+      // Then
+      validateBudleExec(execSpy, gemfilePath)
+      await validateGemFileContent(gemfilePath, isBundled, isWindows)
     })
+  })
 
-    // Then
-    expect(execSpy).toHaveBeenLastCalledWith('bundle', ['exec', 'shopify', 'args'], {
-      stdio: 'inherit',
-      cwd: './directory',
-      env: {
-        ...process.env,
-        SHOPIFY_CLI_STOREFRONT_RENDERER_AUTH_TOKEN: undefined,
-        SHOPIFY_CLI_ADMIN_AUTH_TOKEN: undefined,
-        SHOPIFY_CLI_STORE: undefined,
-        SHOPIFY_CLI_AUTH_TOKEN: 'token_0000_1111_2222_3333',
-        SHOPIFY_CLI_RUN_AS_SUBPROCESS: 'true',
-        BUNDLE_GEMFILE: `/manual/Gemfile`,
-      },
+  it('when run embedded CLI2 in non windows then gemfile content should be correct and bundle runs with correct params', async () => {
+    await file.inTemporaryDirectory(async (cli2Directory) => {
+      // Given
+      const execSpy = await mockEmbeddedCLI2(cli2Directory, !isWindows)
+      const gemfilePath = joinPath(cli2Directory, 'Gemfile')
+
+      // When
+      await execCLI2(['args'], {
+        token: 'token_0000_1111_2222_3333',
+        directory: './directory',
+      })
+
+      // Then
+      validateBudleExec(execSpy, gemfilePath, joinPath(cli2Directory, 'bin', 'shopify'))
+      await validateGemFileContent(gemfilePath, !isBundled, !isWindows)
```

Comment:
When I read this line:
```js
await validateGemFileContent(gemfilePath, !isBundled, !isWindows)
```

I can't tell what the value of `isBundled` and `isWindows` are. I saw them at the root of the module, `const isBundled = true`, and thought you added them there because you'd change the value throughout the various test scenarios, but it turns out you don't.

I'd recommend iterating on your approach to bring clarity. This is useful for your future self, for reviewers, and for contributors coming across this piece of code in the future. If we compare the above with:

```js
await validateGemFileContent(gemfilePath, {
  windows: true,
  isBundled: false,
})
```

I can immediately tell what information is being passed, and what represents.

Reply:
I'm completely agree with you. My intention was to be a `declarative` way of mocking or validating, but your suggestion is really better